File: app/src/main/java/com/example/pratiler/models/Postagem.java
package com.example.pratiler.models;

import java.util.Date;

public class Postagem {
    String texto;
    Leitor leitor;
    Livro livro;
    Integer paginaInicial;
    Integer paginaFinal;
    Date dataPostagem;
    Integer curtidas;
    Boolean curtido;

    public Postagem(String texto, Leitor leitor, Livro livro, Integer paginaInicial, Integer paginaFinal, Date dataPostagem, Integer curtidas, Boolean curtido){
        this.texto = texto;
        this.leitor = leitor;
        this.livro = livro;
        this.paginaInicial = paginaInicial;
        this.paginaFinal = paginaFinal;
        this.dataPostagem = dataPostagem;
        this.curtidas = curtidas;
        this.curtido = curtido;
    }
    public String getTexto() {
        return texto;
    }

    public Leitor getLeitor() {
        return leitor;
    }

    public Livro getLivro() {
        return livro;
    }

    public Integer getPaginaInicial() {
        return paginaInicial;
    }

    public Integer getPaginaFinal() {
        return paginaFinal;
    }

    public Date getDataPostagem() {
        return dataPostagem;
    }

    public Integer getCurtidas() {
        return curtidas;
    }

    public Boolean getCurtido() {
        return curtido;
    }

    public void setTexto(String texto) {
        this.texto = texto;
    }

    public void setLeitor(Leitor leitor) {
        this.leitor = leitor;
    }

    public void setLivro(Livro livro) {
        this.livro = livro;
    }

    public void setPaginaInicial(Integer paginaInicial) {
        this.paginaInicial = paginaInicial;
    }

    public void setPaginaFinal(Integer paginaFinal) {
        this.paginaFinal = paginaFinal;
    }

    public void setDataPostagem(Date dataPostagem) {
        this.dataPostagem = dataPostagem;
    }

    public void setCurtidas(Integer curtidas) {
        this.curtidas = curtidas;
    }

    public void setCurtido(Boolean curtido) {
        this.curtido = curtido;
    }
}

File: hibrido/lib/models/postagem.dart
import 'livro.dart';
import 'leitor.dart';

class Postagem {
  final int id;
  final String texto;
  final Leitor leitor;
  final Livro livro;
  final DateTime data;
  final bool curtido;
  final int curtidas;
  final int paginaInicial;
  final int paginaFinal;

  Postagem({
    required this.id,
    required this.texto,
    required this.leitor,
    required this.livro,
    required this.data,
    required this.curtido,
    required this.curtidas,
    required this.paginaInicial,
    required this.paginaFinal,
  });

  static Postagem fromJson(Map<String, dynamic> json) {
    return Postagem(
      id: json['id'],
      texto: json['texto'],
      leitor: Leitor.fromJson(json['leitor']),
      livro: Livro.fromJson(json['livro']),
      data: DateTime.parse(json['data_hora']),
      curtido: json['curtido'],
      curtidas: json['curtidas'],
      paginaInicial: json['pagina_inicial'],
      paginaFinal: json['pagina_final'],
    );
  }

  Map<String, dynamic> toSql() {
    return { 'id': this.id, 'texto': this.texto, 'leitor': this.leitor.id, 'livro': this.livro.id, 'data': this.data, 'curtido': this.curtido, 'curtidas': this.curtidas, 'paginaInicial': this.paginaInicial, 'paginaFinal': this.paginaFinal }
  }
}
